### PR TITLE
Vk validation fixes, plus PowerVR swapchain hack

### PR DESCRIPTION
**Pull request by hrydgard**
This makes us at least boot to a workable menu on PowerVR so you can switch out of Vulkan into OpenGL. However the emulator is still mostly broken, which I attribute to driver bugs at this point. It's also very slow, both in OpenGL and what works of Vulkan, but maybe Honor 7S is just a bad device (#11743).

It's possible that this could help #11457 too.

### CODE REVIEW

**Review by unknownbrackets — APPROVED** [review]
Looks like SDL needs updating.

I assume width/height still update on rotate for Android?

-[Unknown]

**Comment by hrydgard** [comment]
I'll fix up SDL. Android rotation works just fine.

**Comment by unknownbrackets** [comment]
Maybe it will even fix the Linux/SDL resizing issue somehow...

-[Unknown]

**Comment by hrydgard** [comment]
It doesn't. I have a separate branch with an attempt for that which I'll revive soon...

File: Common/Vulkan/VulkanImage.cpp [hunk 1]
```diff
@@ -76,11 +76,8 @@ bool VulkanTexture::CreateDirect(VkCommandBuffer cmd, int w, int h, int numMips,
 
 	if (allocator_) {
 		offset_ = allocator_->Allocate(mem_reqs, &mem_, Tag());
-		if (offset_ == VulkanDeviceAllocator::ALLOCATE_FAILED) {
-			vkDestroyImage(vulkan_->GetDevice(), image_, nullptr);
-			ELOG("Image memory allocation failed (mem_reqs.size = %d)", (int)mem_reqs.size);
-			return false;
-		}
+		// Destructor will take care of the image.
+		return false;
```
Comment (unknownbrackets) [inline Common/Vulkan/VulkanImage.cpp]: Wait, shouldn't this be inside the ALLOCATE_FAILED if?

-[Unknown]

Comment (hrydgard) [reply Common/Vulkan/VulkanImage.cpp]: Oops what happened here.. sloppy editing. I'll fix soon.

**Pull request merged**